src/pb/publisher.rs

```diff
@@ -2,45 +2,131 @@
 // SPDX-FileCopyrightText: © Contributors to the iceoryx-rs project
 // SPDX-FileContributor: Mathias Kraus
 
-use super::{sample::SampleMut, Topic, POD};
-use crate::IceOryxError;
+use super::{ffi::Publisher as FfiPublisher, sample::SampleMut, PublisherOptions, POD};
+use crate::ConsumerTooSlowPolicy;
+use crate::IceoryxError;
+
+use std::marker::PhantomData;
+
+pub struct PublisherBuilder<'a, T: POD> {
+    service: &'a str,
+    instance: &'a str,
+    event: &'a str,
+    options: PublisherOptions,
+    phantom: PhantomData<T>,
+}
+
+impl<'a, T: POD> PublisherBuilder<'a, T> {
+    pub fn new(service: &'a str, instance: &'a str, event: &'a str) -> Self {
+        Self {
+            service,
+            instance,
+            event,
+            options: PublisherOptions::default(),
+            phantom: PhantomData,
+        }
+    }
+
+    pub fn history_capacity(mut self, history_capacity: u64) -> Self {
```

Comment:
Isn't there something like with C++ a macro which handles the assignment and default value so that we only have to define the option with type and default value and the remaining stuff is generated?

Reply:
You mean something like this https://crates.io/crates/derive_builder?

Reply:
exactly, please use it :)

Reply:
Using macros has its prize since it slows down compilation. For the time being I would keep the manual approach and maybe switch to the macro at a later point